packages/@aws-cdk/aws-rds/lib/cluster-engine.ts

```diff
@@ -23,6 +23,13 @@ export interface ClusterEngineBindOptions {
    */
   readonly s3ExportRole?: iam.IRole;
 
+  /**
```

Comment:
Looking at this holistically. I'm wondering if we should deprecate the props for each role type and, instead, have a prop that something like `addRoles` (not that, that's probably a bad prop name), that takes in an array of roles to add. Those roles existing in a enum like class similar to `Schedule` in `aws-events`.  That feels like a tighter contract and a potentially better user experience. What do you think?

Reply:
I would be fine with that. I was worried about backwards compatibility and this is my first contribution, so I mostly tried to keep it aligned with how the s3 roles work. It would require a wrapper or something to specify which specific feature/parameter the role is for. It relates to my below response, but the roles must be attached by defining what they are for.

For example, in a MySQL Aurora cluster, the role meant for lambda integration has to be attached as the `aws_default_lambda_role` ([link](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraMySQL.Integrating.Lambda.html#AuroraMySQL.Integrating.LambdaAccess)). For S3 export Integration, that parameter is `aurora_select_into_s3_role` or ` aws_default_s3_role` ([link](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraMySQL.Integrating.SaveIntoS3.html#AuroraMySQL.Integrating.SaveIntoS3.Authorize))

I am having some difficulty seeing how the enum class works. Can you elaborate?